iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/truncator/DocumentMetadataAvroTruncator.java

```diff
@@ -0,0 +1,200 @@
+package eu.dnetlib.iis.wf.importer.infospace.truncator;
+
+import eu.dnetlib.iis.importer.schemas.Author;
+import eu.dnetlib.iis.importer.schemas.DocumentMetadata;
+import eu.dnetlib.iis.wf.importer.infospace.ImportInformationSpaceConstants;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of {@link AvroTruncator} for {@link DocumentMetadata}.
+ */
+public class DocumentMetadataAvroTruncator implements AvroTruncator<DocumentMetadata> {
+    private AbstractTruncator abstractTruncator =
+            new AbstractTruncator(ImportInformationSpaceConstants.MAX_ABSTRACT_LENGTH);
+    private TitleTruncator titleTruncator =
+            new TitleTruncator(ImportInformationSpaceConstants.MAX_TITLE_LENGTH);
+    private AuthorsTruncator authorsTruncator =
+            new AuthorsTruncator(ImportInformationSpaceConstants.MAX_AUTHORS_SIZE, ImportInformationSpaceConstants.MAX_AUTHOR_FULLNAME_LENGTH);
+    private KeywordsTruncator keywordsTruncator =
+            new KeywordsTruncator(ImportInformationSpaceConstants.MAX_KEYWORDS_SIZE, ImportInformationSpaceConstants.MAX_KEYWORD_LENGTH);
+
+    @Override
+    public DocumentMetadata truncate(DocumentMetadata source) {
+        if (Objects.nonNull(source.getAbstract$())) {
+            source.setAbstract$(abstractTruncator.truncate(source.getAbstract$()));
+        }
+
+        if (Objects.nonNull(source.getTitle())) {
+            source.setTitle(titleTruncator.truncate(source.getTitle()));
+        }
+
+        if (Objects.nonNull(source.getAuthors())) {
+            source.setAuthors(authorsTruncator.truncate(source.getAuthors()));
+        }
+
+        if (Objects.nonNull(source.getKeywords())) {
+            source.setKeywords(keywordsTruncator.truncate(source.getKeywords()));
+        }
+
+        return source;
+    }
+
+    public static class AbstractTruncator implements Serializable {
+        private Function<CharSequence, CharSequence> stringTruncator;
+
+        public AbstractTruncator(int maxAbstractLength) {
+            this.stringTruncator = (Function<CharSequence, CharSequence> & Serializable) x ->
+                    StringTruncator.truncateWithoutWordSplit(x.toString(), maxAbstractLength);
+        }
+
+        public void setStringTruncator(Function<CharSequence, CharSequence> stringTruncator) {
+            this.stringTruncator = stringTruncator;
+        }
+
+        CharSequence truncate(CharSequence x) {
+            return stringTruncator.apply(x);
+        }
+    }
+
+    public static class TitleTruncator implements Serializable {
```

Comment:
Do we really need a dedicated Truncator instance for each field? `AbstractTruncator` looks exactly the same as the `TitleTruncator`. 

I am not sure we really need to propagate the truncators model deep down below the `DocumentMetadataAvroTruncator` for each field. I know there are the test classes for each case, which is nice, but maybe we could live with just a truncation test for `DocumentMetadata` and test truncation of all the dimensions. There are only 6 of them. I'd just apply whole truncation logic within the:
```
public DocumentMetadata truncate(DocumentMetadata source)
```
method and drop all the internal truncators. To me it feels overcomplicated.

Reply:
Done, I've removed dedicated truncator classes for each field. The only reason for their implementation was to make unit testing cleaner.